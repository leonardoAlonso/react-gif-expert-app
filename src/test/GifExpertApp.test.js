import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';

import { GifExpertApp } from '../GifExpertApp'

describe("Test in GifExperApp", () => {

    test("debe existir GifExperApp", () => {

        const wrapper = shallow(<GifExpertApp/>)
        expect( wrapper ).toMatchSnapshot()

    })

    test('debe de mostrar una lista de categorias ', () => {
        const categories = ['Naruto', 'DragonBall'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length)
    })
    

});