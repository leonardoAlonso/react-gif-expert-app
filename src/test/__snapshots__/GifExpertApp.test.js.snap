// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExperApp debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="DragonBall"
      key="DragonBall"
    />
  </ol>
</Fragment>
`;

exports[`Test in GifExperApp debe existir GifExperApp 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
